src/tarball.rs

```diff
@@ -39,10 +39,29 @@ impl diff::Diff for LockedTarball {
     }
 }
 
+// HACK: this is silly, but we need this because we `serde::flatten` our pins, so we need a struct
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct HashWrapper {
```

Comment:
I don't really understand this type. What is the difference to the previous `GenericHash` type? This change reads like going full circle to me

Reply:
I mean, yeah, pretty much :/

I could've sworn that using `NixHash` directly causes a serde failure due to `#[serde(flatten)]`, but I've just tried locally and can't reproduce? Idk what I've stumbled into before, but yeah, I'm pushing a good version in a sec.